test: add unit tests for tempo.js

Inuth0603 opened the pull request:
Adds unit tests for the Tempo widget logic (initialization, speedUp, slowDown, limits).

Fixes #4973
<img width="855" height="333" alt="Screenshot 2026-01-01 131253" src="https://github.com/user-attachments/assets/d22b770d-edfc-4a00-8481-2be6f8b9d594" />

Jetshree commented:
Please add license header whenever you are creating a new file. 
Check for #4972 too.

Jetshree commented:
@Inuth0603 can you please share coverage report again?

Inuth0603 commented:
Apologies for the delay in replying. I have added the license headers to the new files in this PR and also updated PR #4972 as requested.
<img width="1920" height="1080" alt="Screenshot 2026-01-01 125041" src="https://github.com/user-attachments/assets/7b1273c1-2426-4307-812d-0d64907f1693" />
<img width="1920" height="1080" alt="Screenshot 2026-01-01 125421" src="https://github.com/user-attachments/assets/005cd500-9602-4338-8485-ce5930279dab" />
Also the screenshot in my description reflects a local isolated run (npx jest js/widgets/tempo.test.js) where I only executed this specific test file to verify my changes. That is why the global coverage for other files shows as 0%. The coverage for tempo.js itself is the relevant part here.

Jetshree commented:
> Apologies for the delay in replying. I have added the license headers to the new files in this PR and also updated PR #4972 as requested. <img alt="Screenshot 2026-01-01 125041" width="1920" height="1080" src="https://private-user-images.githubusercontent.com/218235267/531312307-7b1273c1-2426-4307-812d-0d64907f1693.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjcyNTI3NDEsIm5iZiI6MTc2NzI1MjQ0MSwicGF0aCI6Ii8yMTgyMzUyNjcvNTMxMzEyMzA3LTdiMTI3M2MxLTI0MjYtNDMwNy04MTJkLTBkNjQ5MDdmMTY5My5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjYwMTAxJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI2MDEwMVQwNzI3MjFaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1kOTM4ZTBmYmUxMzlkMzJmNjhhYmMzZGJmODNhYmUyYzc3YzMwMTZhMTJkZTU1MTdlMGE1MTI3ZjNiNzZiMjU5JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.iROCUl9anhDissYVya8CCjZlfDkfMl-ayvZdFIFckJI"> <img alt="Screenshot 2026-01-01 125421" width="1920" height="1080" src="https://private-user-images.githubusercontent.com/218235267/531312429-005cd500-9602-4338-8485-ce5930279dab.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjcyNTI3NDEsIm5iZiI6MTc2NzI1MjQ0MSwicGF0aCI6Ii8yMTgyMzUyNjcvNTMxMzEyNDI5LTAwNWNkNTAwLTk2MDItNDMzOC04NDg1LWNlNTkzMDI3OWRhYi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjYwMTAxJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI2MDEwMVQwNzI3MjFaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT03NDk1Y2NkODgxZTVmNGNhOTQxOGU2YjAyM2RhY2U4YmIzOTc0OTkzMzViOTExMmRiZTZmYmUzZDFkOTMzNGM4JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.lvvBmoBV00d1MbASk1BxjjR4JlHt09UmS5bh2BadhdA"> Also the screenshot in my description reflects a local isolated run (npx jest js/widgets/tempo.test.js) where I only executed this specific test file to verify my changes. That is why the global coverage for other files shows as 0%. The coverage for tempo.js itself is the relevant part here.

Okay, still i suggest to add ss focusing on coverage of the file you have changed. This makes your changes easy to review.
Please update the image in PR description so that coverage of tempo widget is clearly visible.

Inuth0603 commented:
I have updated the PR description with the focused coverage screenshot for tempo.js. Thanks!
